internal/server/kong/ws/config/compat/compatibility_table.go

```diff
@@ -834,6 +842,20 @@ var (
 				},
 			},
 		},
+		{
+			Metadata: config.ChangeMetadata{
+				ID:          config.ChangeID("P135"),
```

Comment:
We'll want to get in a test for this.

Reply:
Good catch on this test, I'm pretty sure that core entity deletion hasn't actually been implemented.

Reply:
It is, you will just need to add a new vaults type; see [here](https://github.com/Kong/koko/blob/738960912041441f287456f1fdbeefe58da43cfc/internal/server/kong/ws/config/version_compatibility.go#L32-L51). From here you will want to add an [entry to the compatibility table along with change metadata information](https://github.com/Kong/koko/blob/738960912041441f287456f1fdbeefe58da43cfc/internal/server/kong/ws/config/compat/compatibility_table.go#L86) and then [this entity removal loop](https://github.com/Kong/koko/blob/738960912041441f287456f1fdbeefe58da43cfc/internal/server/kong/ws/config/version_compatibility.go#L521-L542) will take over the action.

Reply:
@mikefero this isn't a field removal, it's removing the entire vaults entity.

Reply:
https://github.com/Kong/koko/pull/464/files#diff-ca9bb92e11fab1c215299a0598bbd953852f976044f0bc8cd18cd8a0b9912d50R264-R302

Some extra eyes on this part would be great, I am not super familiar with this part of the code, but I think I have it patched together.

Reply:
Nice, I just added some comments.

On top of those:
- can you add some test for this in `version_compatibility_test.go`? (as pointed out by Taylor)
- can you also add some e2e test [here](https://github.com/Kong/koko/blob/main/internal/test/e2e/version_compatibility_test.go) to make sure this entity is remove for DP < 3.0.0?

Reply:
> @mikefero this isn't a field removal, it's removing the entire vaults entity.

:man_facepalming: You are 100% correct.

In that case we replicate [removePlugin](https://github.com/Kong/koko/blob/738960912041441f287456f1fdbeefe58da43cfc/internal/server/kong/ws/config/version_compatibility.go#L266-L306) functionality except do it for entities. I would say that is kind of feature creep for this PR and would be better suited in a separate one.

Reply:
#483